lib/kafka_producer.js

```diff
@@ -157,10 +244,54 @@ KafkaProducer.prototype.getWholeMsg = function getWholeMsg(topic, message, timeS
 
 KafkaProducer.prototype.close = function close(callback) {
     var self = this;
+    if (self.batching) {
+        self.flushEntireCache(callback);
+    }
     self.enable = false;
     if (self.restClient) {
         self.restClient.close();
     }
 };
 
+// The batch message payload should follow:
+// 4 BE bytes number of messages + 4 BE bytes size of message + actual message
+function MessageBatch(topic) {
```

Comment:
I'd really recommend putting this thing in another file so you can unit test it... you are going to unit test it, right? ;-)